enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,130 @@
+---
+title: allow-olm-operators-to-request-a-namespace-to-be-installed-into-and-enable-cluster-monitoring-on-that-namespace
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-12-11
+last-updated: 2019-12-19
+status: implementable
+---
+
+# allow-olm-operators-to-request-a-namespace-to-be-installed-into-and-enable-cluster-monitoring-on-that-namespace
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary and Motivation
+
+The primary purpose of this enhancement is to propose a series of changes that would enable [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) managed operators to report metrics to the [OpenShift Monitoring](https://github.com/openshift/cluster-monitoring-operator) [Prometheus Operator](https://github.com/coreos/prometheus-operator).
+
+First, OLM will enable operators to interact with a Prometheus Operator deployed on vanilla Kubernetes clusters. This support will be implemented by allowing OLM managed operators to package `ServiceMonitor` and `PrometheusRule` objects within their bundle. OLM will then manage these resources alongside the lifecycle of the operator, as OLM currently does with a number of other resources.
+
+Second, OLM will define how an operator author can:
+
+- Suggest a namespace that the operator should be deployed to
+- Specify that the operator should be scraped by the OpenShift Monitoring Prometheus Operator present on all OpenShift clusters.
+
+This information will be consumed by [OpenShift Console UI](https://github.com/openshift/console) to integrate the operator into the OpenShift Monitoring Prometheus Operator.
+
+### Goals
+
+- Provide OLM managed operators with a first class way of creating `ServiceMonitors` and `PrometheusRules` resources to interact with the Prometheus Operator
```

Comment:
Should we support PodMonitor as well?

Reply:
Based on the [Prometheus API Documentation](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#podmonitorspec), PodMonitors are a lot like ServiceMonitors but they're targeted at Pods.